### DISCUSSION TITLE
getRemainingTime updated when none of the methods are called

### DISCUSSION THREAD

**Discussion by Josh68** [SupremeTechnopriest/react-idle-timer#340] [Q&A]
Hello. I've tried debugging this locally, and haven't figured out what's going on. Before trying to post any code (which is pretty complex, and inolves an external session API), I just want to make sure I'm understanding things.

I now have listeners set up on every event/state-change method of the timer (I'm using the hook), and as in the demos, I'm currently logging out remaining time in the UI so I can see what's happening. I also have both `startOnMount` and `startManually` set to `false`, and control the timer by starting it only when a user has an authenticated session and updating it by setting values for `timeout` and `promptBeforeIde` in the wrapping component's `state`.

Sometimes when I move my mouse or similar, or when I tab away for a bit and come back, I see `onAction` being called with the relevant event (like `mousemove` or `visibilitychange`) . This is expected. In these cases, I declaratively update my time values, and then expect my timer to restart and for `getRemainingTime` to be updated, with more time remaining.

But there are also times when moving my mouse or tabbing does not trigger any callback, and still `getRemainingTime` is updated. Here I mean I have callbacks for `onIdle`, `onPrompt`, `onAction`, `onActive` (which still confuses me, vs `onAction`), `onMessage`, and `onPresenceChange`.

Why would `getRemainingTime` be updated with more time on something like a `mousemove` if none of the timer's event or state-change callbacks have been called?

Thanks

**Comment by Josh68** [comment]
I have resorted to working around this issue by setting my `events` to an empty array, then creating a new `useEffect` hook that binds to all `DEFAULT_EVENTS`, with throttling, and then calls my existing `onActionHandler`. I'd love to not need to do this, but can't figure out why my timer's remaining time was updating when none of the event/state-change methods were being called, and particularly `onAction`.

**Comment by Josh68** [comment]
I now think my issue was using `throttle` instead of `eventsThrottle` to control how often the idle timer's interal timeouts would be updated. This is another thing I find a little confusing about the API.

**Reply by SupremeTechnopriest** [reply]
Yeah I agree `eventsThrottle` is not a great name for this property.  I am open to suggestions.
